python/lsst/ts/ofc/ofc_data/ofc_data.py

```diff
@@ -350,122 +314,207 @@ def _configure_instrument(self, value):
-        izk_filenames = [
-            Path(_p) for _p in glob(str(inst_config_dir / intrinsic_zk_pattern))
-        ]
-
-        self.log.debug(f"Configuring instrisic zernikes: {len(izk_filenames)} files.")
+        # If the camera type is lsst read and set up
+        # y2_correction for the gaussian quadrature points
+        # ------------------------------------------------
+        if instrument == "lsst":
+            gq_y2_path = (
+                self.config_dir
+                / "y2"
+                / f"{instrument}_gaussian_quadrature{self.y2_filename_root}"
+            )
 
-        intrinsic_zk = dict()
+            self.log.debug(f"Configuring y2: {gq_y2_path}")
 
-        for filename in izk_filenames:
-            filter_name_regex = _INTRISICZN_FILTER_REGEX.match(filename.name)
-            if filter_name_regex is None:
-                RuntimeError(
-                    f"Could not parse filter for intrinsic zernike coefficient file {filename.name}. "
-                    "Either remove or rename the file so it does not match the pattern "
-                    f"{intrinsic_zk_pattern}."
+            try:
+                with open(gq_y2_path) as fp:
+                    gq_y2_correction = yaml.safe_load(fp)
+            except FileNotFoundError:
+                raise RuntimeError(
+                    "Could not read y2 gaussian quadrature file "
+                    f"from instrument config directory: {gq_y2_path!s}. "
+                    "Check your instrument configuration directory integrity."
                 )
 
-            filter_name = filter_name_regex["name"]
+        # Read all intrinsic zernike coefficients data
+        # --------------------------------------------
+        self.log.debug(
+            f"Configuring instrisic zernikes: {len(self.eff_wavelength.keys())} files."
+        )
 
-            if filter_name in intrinsic_zk:
-                RuntimeError(
-                    f"Filter name {filter_name} already accounted for. "
-                    f"Make sure the files in {inst_config_dir} does not duplicate filter names."
-                )
+        intrinsic_zk = dict()
+        intrinsic_zk_path = self.config_dir / "intrinsic_zernikes" / camera_type
+        if not intrinsic_zk_path.exists():
+            RuntimeError(
+                f"Instrument {camera_type!s} intrinsic zernikes directory does not exists. "
+                f"Make sure the name you are passing ({camera_type}) "
+                "exists in the intrinsic zernikes directory"
+            )
 
-            with open(filename) as fp:
-                intrinsic_zk[filter_name] = np.array(yaml.safe_load(fp))
+        for filter_name in self.eff_wavelength.keys():
+            file_name = (
+                f"{self.intrinsic_zk_filename_root}_{filter_name.lower()}_31*.yaml"
+            )
+            intrinsic_file = Path(glob(str(intrinsic_zk_path / file_name))[0])
```

Comment:
I think the `pathlib` has the similar function to replace the `glob()`.  I will use it because you used the `pathlib` already.